### PR TITLE
fix(login): Make login form and footer look like a dashboard panel

### LINKED ISSUE #36950: [BITV] 9.1.4.3/1.2 - When using a user stylesheet (html {background-color:black;color:white}), the sentence "Nextcloud - a secure place for all your data" at the bottom of the page (white text) is no longer properly readable/discernible. (1)
on login page 

- [ ] Fix
- [ ] Backport

image

<details>
<summary>Details</summary>

https://report.bitvtest.de/default-en/d63601ac-cb34-4645-8256-66bec78964a0.html#checkpoint-6fe46b78da-v1-n2

</details>

### PR DESCRIPTION
**Pull request by susnux**
* Resolves: #36950

## Summary
1. Make the background of the login form look like a dashboard panel.
2. Add the same background style to the footer so it is independent of the background image / color and fulfill the contrast requirements.

## Screenshots

what | before | after
---|---|---
shot footer | Screenshot 2023-05-09 at 00-36-55 Nextcloud|image
long footer | Screenshot 2023-05-09 at 00-38-35 Nextcloud|image
`color: white; background: black;`¹ |Screenshot 2023-05-09 at 00-37-26 Nextcloud|image

¹: Like the issue: if the user agent uses a custom default browser theme

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Screenshots before/after for front-end changes

### CODE REVIEW

**Comment by solracsf** [comment]
IMO, the same type or border radius should be applied.

image

**Comment by susnux** [comment]
@szaimen @solracsf there was a design discussion in the linked issue and also here https://github.com/nextcloud/server/issues/35482#issuecomment-1534723358, so I just picked the pill style as it seems to be the preferred one (?).

> maxbe also make it the same width like the login background?

I do not think that works, because the footer text is much wider, looks like this:

image

**Comment by solracsf** [comment]
No problem, but from a design perspective, it makes no sense (to me) to have multiple border radius ;-)

**Comment by nimishavijay** [comment]
I would vote for pill style too, so that it reflects how it looks in the dashboard (border-radius-large boxes, and the smaller buttons are pill shaped). What do you think? And agreed about the width, it might make the height too big, especially for translations. 

What do you think about the existing padding being reduced such that the height of the div is 44px? It could look like this:

**Comment by szaimen** [comment]
with reduced padding it indeed looks much better 👍

**Comment by susnux** [comment]
@nimishavijay is it very important that the footer is not wider than the login box?
Because for English it works, but most other languages I tried (French, Spanish, Portuguese, German) require more space (about 10 to 30px more).

I personally think that less height but wider footer look better:

Spanish (`min-width: 300px`) | Portuguese (`width: 300px`)
---|---
image | image

**Comment by nimishavijay** [comment]
> Because for English it works, but most other languages I tried (French, Spanish, Portuguese, German) require more space 

Oh it was a misunderstanding! I agreed with your point about the width, so it doesn't need to be limited :)

**Review by nimishavijay — APPROVED** [review]
Latest screenshots look great! Good to go design-wise :) 🚀

**Review by jancborchardt — APPROVED** [review]
Looks great @susnux! :) I’d actually say we could use an even larger border-radius for the login box as well – the same as for the content when logged in: `var(--body-container-radius)`. What do you think?

**Comment by susnux** [comment]
> I’d actually say we could use an even larger border-radius for the login box as well – the same as for the content when logged in: `var(--body-container-radius)`. What do you think?

@jancborchardt 
I intended to make it consistent with the dashboard panels, which use `--border-radius-large`:
https://github.com/nextcloud/server/blob/9d2d3d482bce004d134dd87d7b37c3dfb6513412/apps/dashboard/src/DashboardApp.vue#L416

As the login box is more a panel than the main content box (?)

**Comment by jancborchardt** [comment]
Right, that makes sense @susnux – considering that the Dashboard panels are older than the redesign though, I’d say it’s better to align with the redesign. :) Especially since the new added container around the footer looks rounder than the login form container otherwise.

And actually the dashboard panels also look better with the bigger border-radius (`var(--body-container-radius)` corresponds to 28px)

Login screen with bigger border radius | Dashboard panel with bigger border radius
-|-
image|image

File: core/src/views/Login.vue [hunk 1]
```diff
@@ -203,6 +205,34 @@ body {
...
+.login-box.guest-box, footer {
+	color: var(--color-main-text);
+	background-color: var(--color-main-background-blur);
+	-webkit-backdrop-filter: var(--filter-background-blur);
+	backdrop-filter: var(--filter-background-blur);
+}
+
+footer {
+	// Usually the same size as the login box, but allow longer texts
+	min-width: 320px;
```
[the remark this answers is not in the document]
Comment (susnux) [reply core/src/views/Login.vue:219]: Yes, seems to work fine, even on quite low resoultion (350px width) it looks ok.

For example the login page on 350px:
English (medium size text) | Portuguese (long text)
---|---
image|image

**Review by szaimen — CHANGES_REQUESTED** [review]
I just tested this and it needs a fix for mobile:

image

File: core/src/views/Login.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (szaimen) [reply core/src/views/Login.vue:219]: As you can see is this overlapping on the left and right but fine by me...

**Comment by susnux** [comment]
@szaimen yes, cache issue... Happens because that css variable is set to `0px` for mobile view. Fixed it.

File: core/src/views/Login.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (susnux) [reply core/src/views/Login.vue:219]: Fixed for mobile

**Review by szaimen — APPROVED** [review]
Tested and seems to work now but didnt review the code

**Review by jancborchardt — APPROVED** [review]
Seems good to me! And now looks very nice with the 2 similar roundings. :) Nice job @susnux

**Comment by szaimen** [comment]
CI failure unrelated

**Pull request merged**